File: RiskRoller.py
#!/usr/bin/python3

'''
Risk roller is a calculator for working out who wins a round in a game of risk
'''
import random

def dice_roll(number_of_dice): #returns a list of randomly generated numbers between 1 and 6, for a specified number of dice, sorted highest to lowest
	result = []
	for n in range(number_of_dice):
		result.append(random.randint(1,6))

	result.sort(reverse = True)

	return (result)

def attack(attacker_armies, defender_armies,attack_stop,attacker_number_of_dice):
	attack_round = 1


	while attacker_armies > attack_stop and defender_armies > 0:
		# rolls  dice for attacker
		if attacker_armies >= 3:
			attacker_dice = dice_roll(3)
		elif attacker_armies == 2 or attacker_number_of_dice == 2:
			attacker_dice = dice_roll(2)
		elif attacker_armies == 1 or attacker_number_of_dice == 1:
			attacker_dice = dice_roll(1)

		# rolls 2 dice if attacker has >2 armies
		if defender_armies >= 2:
			defender_dice = dice_roll(2)	
		# else rolls 1 die
		else :
			defender_dice = dice_roll(1)

		print('ROUND:', attack_round, 'Attacker has', attacker_armies, 'armies. Defenders has', defender_armies,'armies.')
		print("Attacker Dice = ", attacker_dice)	
		print("Defender Dice = ", defender_dice)

		# calculates who wins for each roll
		defender_lose = 0
		attack_lose = 0
		for i in range(len(defender_dice)):		
			if attacker_dice[i] > defender_dice[i]:
				#attacker wins
				defender_lose += 1
				defender_armies -= 1
			else:
				#defender wins
				attack_lose += 1
				attacker_armies -= 1
		print('Defender loses: ', defender_lose , ' Attacker loses: ', attack_lose)
		attack_round += 1

	return{'round':attack_round, 'attacker_armies':attacker_armies, 'defender_armies': defender_armies}



def main():	
	#get initial number of armies
	attacker = int(input('Number of attacking armies:'))
	defender = int(input('Number of defending armies:'))

	attack_result = attack(attacker, defender,3,3)

	print('\n#####################\nAfter:',attack_result['round'],'rounds.') #LINE BREAK
	
	if attack_result['attacker_armies'] <=3 :
		print('DEFENDER WINS. Attacker has', attack_result['attacker_armies'], 'armies remaining. Defender has', attack_result['defender_armies'],'armies remaining.' )

	elif attack_result['defender_armies'] == 0:
		print('ATTACKER WINS: Attacker has', attack_result['attacker_armies'], 'armies remaining' )
	else:
		print('something\'s gone wrong, play Settlers of Catan instead.')
	print('#####################')

if __name__ == '__main__':
	main()

File: riskRoller.js
//define a random number generator that can make a list of random numbers between 1 & 6

function getRandomInt(min, max) {
  min = Math.ceil(min);
  max = Math.floor(max);
  return Math.floor(Math.random() * (max - min)) + min; //The maximum is exclusive and the minimum is inclusive
}

var diceRoll = (number_of_dice) => {
	var result = [];
	for (var n = 0; n < number_of_dice; n++){
		result.push(getRandomInt(1,7))
	}
	result.sort(function(a,b){return b-a});

	return result;
}

console.log(diceRoll(3));
document.write(diceRoll(3));

//get the intital number of armies
//set as defined values for now

let attacker = 9;
let defender = 9;


//while loop for normal conditions

let game_round = 0;
while (attacker > 3 && defender >0){
	attacker_dice = diceRoll(3);

	if(defender >=2){
		defender_dice = diceRoll(2);
	} else{
		defender_dice = diceRoll(1);
	}
}
	//rolls 3 dice for the attacker

	//rolls 2 dice if the defender has >2 armies

	//else rolls 1 dice

	//calculates win/loss for each roll
		//attacker wins

		//defender wins


//If defender = 0 or attacker = 3, outputs the response
